File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/StringSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface StringSchemaValidator<T> {
    String validate(String arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
    T validateAndBox(String arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/StringSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface StringSchemaValidator<T> {
    String validate(String arg, SchemaConfiguration configuration) throws ValidationException;
    T validateAndBox(String arg, SchemaConfiguration configuration) throws ValidationException;
}